test/metric/collectd.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+type CollectDMetricValueFetcher struct {
+	baseMetricValueFetcher
+}
+
+var _ MetricValueFetcher = (*CollectDMetricValueFetcher)(nil)
+
+func (f *CollectDMetricValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
```

Comment:
This seems common for all plugins... does this ever differ for any plugin?
Same question for `isApplicable`

Reply:
I agree that they're redundant. I haven't put a lot of thought into how to consolidate, but I think that `isApplicable` is the easier one to clean up since we already have a list of supported metrics that we have to implement in the fetcher interface. We _should_ be able to just assume that `isApplicable` is consistent across all implementations.